### PR TITLE
[FLINK-20546][kafka]fix method misuse in DynamicTableFactoryTest

### PR DESCRIPTION
**Pull request by zoucao**
## What is the purpose of the change
fix method misuse in `KafkaDynamicTableFactoryTest` and `UpsertKafkaDynamicTableFactoryTest`.

## Brief change log

*(for example:)*
  - use correct method to get options for creating table.

## Verifying this change

*(Please pick either of the following options)*

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactoryTest.java [hunk 1]
```diff
@@ -585,7 +585,6 @@ public void testPrimaryKeyValidation() {
 					"I;UA;UB;D"));
 		// pk can be defined on cdc table, should pass
 		createTableSink(pkSchema, options1);
-		createTableSink(pkSchema, options1);
```
Comment (fsk119) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactoryTest.java:588]: It should be `createTableSource(...)`.

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I left a minor comment.

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactoryTest.java (hunk 1 above)
Comment (zoucao) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactoryTest.java:588]: I just thought it was a repeat call before your comment, thanks for your comment, I will fix it soon.

**Review by fsk119 — APPROVED** [review]
LGTM

**Pull request merged**